src/main/java/com/mraof/minestuck/block/EnumCassetteType.java

```diff
@@ -1,24 +1,88 @@
 package com.mraof.minestuck.block;
 
+import com.mraof.minestuck.util.MSSoundEvents;
+import net.minecraft.sounds.SoundEvent;
+import net.minecraft.sounds.SoundEvents;
 import net.minecraft.util.StringRepresentable;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.effect.MobEffects;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+import java.util.function.Supplier;
 
 public enum EnumCassetteType implements StringRepresentable
 {
-	NONE,
-	THIRTEEN,
-	CAT,
-	BLOCKS,
-	CHIRP,
-	FAR,
-	MALL,
-	MELLOHI,
-	EMISSARY_OF_DANCE,
-	DANCE_STAB_DANCE,
-	RETRO_BATTLE_THEME;
+	NONE(() -> null, null),
+	ELEVEN(() -> SoundEvents.MUSIC_DISC_11, onHitEffect(() -> new MobEffectInstance(MobEffects.WITHER, 80, 0),
+			0.10F)),
+	THIRTEEN(() -> SoundEvents.MUSIC_DISC_13, onHitEffect(() -> new MobEffectInstance(MobEffects.HUNGER, 100, 0), 0.30F)),
+	BLOCKS(() -> SoundEvents.MUSIC_DISC_BLOCKS, onHitEffect(() -> new MobEffectInstance(MobEffects.BLINDNESS, 100, 0),
+			0.15F)),
+	CAT(() -> SoundEvents.MUSIC_DISC_CAT, userEffect(() -> new MobEffectInstance(MobEffects.NIGHT_VISION, 200, 0,
+			false, false, false))),
+	CHIRP(() -> SoundEvents.MUSIC_DISC_CHIRP, userEffect(() -> new MobEffectInstance(MobEffects.SLOW_FALLING, 100, 0,
+			false, false, false))),
+	DANCE_STAB_DANCE(MSSoundEvents.MUSIC_DISC_DANCE_STAB_DANCE, onHitEffect(() -> new MobEffectInstance(MobEffects.DIG_SLOWDOWN, 500, 3),
+			0.3F)),
+	EMISSARY_OF_DANCE(MSSoundEvents.MUSIC_DISC_EMISSARY_OF_DANCE, onHitEffect(() -> new MobEffectInstance(MobEffects.POISON, 200, 0),
+			0.1F)),
+	FAR(() -> SoundEvents.MUSIC_DISC_FAR, userEffect(() -> new MobEffectInstance(MobEffects.MOVEMENT_SPEED, 100, 0,
+			false, false, false))),
+	MALL(() -> SoundEvents.MUSIC_DISC_MALL, userEffect(() -> new MobEffectInstance(MobEffects.WATER_BREATHING, 100, 0,
+			false, false, false))),
+	MELLOHI(() -> SoundEvents.MUSIC_DISC_MELLOHI, onHitEffect(() -> new MobEffectInstance(MobEffects.LEVITATION, 60, 0),
+			0.20F)),
+	PIGSTEP(() -> SoundEvents.MUSIC_DISC_PIGSTEP, userEffect(() -> new MobEffectInstance(MobEffects.FIRE_RESISTANCE, 100, 0,
+			false, false, false))),
+	RETRO_BATTLE_THEME(MSSoundEvents.MUSIC_DISC_RETRO_BATTLE_THEME, userEffect(() -> new MobEffectInstance(MobEffects.HEALTH_BOOST, 100, 2,
```

Comment:
This health boost effect adds 6 hearts(12 health) to the player. I dont necessarily have an issue with it but I did want to know if it was intended

Reply:
This may complicate how the code works but it may be smarter to make health boosting a weapon attribute modifier instead of a potion effect. An issue with health boost is that every time it is reapplied, the player loses that additional health. Meaning that they are cut back down to their original health every ~3 seconds, which doesnt help them with healing and could be considered a waste of the players hunger